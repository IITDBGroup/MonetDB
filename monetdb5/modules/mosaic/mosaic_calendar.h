/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0.  If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * Copyright 1997 - July 2008 CWI, August 2008 - 2018 MonetDB B.V.
 */

/*
 * (c)2014 author Martin Kersten
 */

#ifndef _MOSAIC_TEMPORAL_
#define _MOSAIC_TEMPORAL_

#include <mal.h>
#include "mal_interpreter.h"
#include "mal_client.h"

#define TEMPORALTHRESHOLD 4192
#define TEMPORALSIZE 256 

bool MOStypes_calendar(BAT* b);
mal_export void MOScreatecalendar(Client cntxt, MOStask task);
mal_export void MOSlayout_calendar(Client cntxt, MOStask task, BAT *btech, BAT *bcount, BAT *binput, BAT *boutput, BAT *bproperties);
mal_export void MOSlayout_calendar_hdr(Client cntxt, MOStask task, BAT *btech, BAT *bcount, BAT *binput, BAT *boutput, BAT *bproperties);
mal_export void MOSadvance_calendar(Client cntxt, MOStask task);
mal_export void MOSskip_calendar(Client cntxt, MOStask task);
mal_export flt  MOSestimate_calendar(Client cntxt, MOStask task);
mal_export void MOScompress_calendar(Client cntxt, MOStask task);
mal_export void MOSdecompress_calendar(Client cntxt, MOStask task);
mal_export str MOSselect_calendar(Client cntxt,  MOStask task, void *low, void *hgh, bit *li, bit *hi, bit *anti);
mal_export str MOSthetaselect_calendar(Client cntxt,  MOStask task, void *val, str oper);
mal_export str MOSprojection_calendar(Client cntxt,  MOStask task);
mal_export str MOSjoin_calendar(Client cntxt,  MOStask task);
#endif /* _MOSAIC_TEMPORAL_ */
