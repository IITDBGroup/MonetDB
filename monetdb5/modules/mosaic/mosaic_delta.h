/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0.  If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * Copyright 1997 - July 2008 CWI, August 2008 - 2018 MonetDB B.V.
 */

/*
 * (c)2014 author Martin Kersten
 */

#ifndef _MOSAIC_DELTA_
#define _MOSAIC_DELTA_

#include <mal.h>
#include "mal_interpreter.h"
#include "mal_client.h"

bool MOStypes_delta(BAT* b);
mal_export void MOSlayout_delta(Client cntxt, MOStask task, BAT *btech, BAT *bcount, BAT *binput, BAT *boutput, BAT *bproperties);
mal_export void MOSadvance_delta(Client cntxt, MOStask task);
mal_export void MOSskip_delta(Client cntxt, MOStask task);
mal_export flt  MOSestimate_delta(Client cntxt, MOStask task);
mal_export void MOScompress_delta(Client cntxt, MOStask task);
mal_export void MOSdecompress_delta(Client cntxt, MOStask task);
mal_export str MOSselect_delta(Client cntxt,  MOStask task, void *low, void *hgh, bit *li, bit *hi, bit *anti);
mal_export str MOSthetaselect_delta(Client cntxt,  MOStask task, void *val, str oper);
mal_export str MOSprojection_delta(Client cntxt,  MOStask task);
mal_export str MOSjoin_delta(Client cntxt,  MOStask task);
#endif /* _MOSAIC_DELTA_ */
