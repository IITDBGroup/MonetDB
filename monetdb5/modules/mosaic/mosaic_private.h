#define PREVVALUEbit(x) ((x) - 1)
#define PREVVALUEbte(x) ((x) - 1)
#define PREVVALUEsht(x) ((x) - 1)
#define PREVVALUEint(x) ((x) - 1)
#define PREVVALUElng(x) ((x) - 1)
#define PREVVALUEoid(x) ((x) - 1)
#ifdef HAVE_HGE
#define PREVVALUEhge(x) ((x) - 1)
#endif
#define PREVVALUEflt(x) nextafterf((x), -GDK_flt_max)
#define PREVVALUEdbl(x) nextafter((x), -GDK_dbl_max)
#define PREVVALUEdate(x) ((x) - 1)
#define PREVVALUEdaytime(x) ((x) - 1)

#define NEXTVALUEbit(x) ((x) + 1)
#define NEXTVALUEbte(x) ((x) + 1)
#define NEXTVALUEsht(x) ((x) + 1)
#define NEXTVALUEint(x) ((x) + 1)
#define NEXTVALUElng(x) ((x) + 1)
#define NEXTVALUEoid(x) ((x) + 1)
#ifdef HAVE_HGE
#define NEXTVALUEhge(x) ((x) + 1)
#endif
#define NEXTVALUEflt(x) nextafterf((x), GDK_flt_max)
#define NEXTVALUEdbl(x) nextafter((x), GDK_dbl_max)
#define NEXTVALUEdate(x) ((x) + 1)
#define NEXTVALUEdaytime(x) ((x) + 1)
