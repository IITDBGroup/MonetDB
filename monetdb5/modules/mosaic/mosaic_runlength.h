/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0.  If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * Copyright 1997 - July 2008 CWI, August 2008 - 2018 MonetDB B.V.
 */

/*
 * (c)2014 author Martin Kersten
 */

#ifndef _MOSAIC_RLE_
#define _MOSAIC_RLE_

#include <mal.h>
#include "mal_interpreter.h"
#include "mal_client.h"

mal_export void MOSdump_runlength(Client cntxt, MOStask task);
mal_export void MOSlayout_runlength(Client cntxt, MOStask task, BAT *btech, BAT *bcount, BAT *binput, BAT *boutput, BAT *bproperties);
mal_export void MOSadvance_runlength(Client cntxt, MOStask task);
mal_export void MOSskip_runlength(Client cntxt, MOStask task);
mal_export flt  MOSestimate_runlength(Client cntxt, MOStask task);
mal_export void MOScompress_runlength(Client cntxt, MOStask task);
mal_export void MOSdecompress_runlength(Client cntxt, MOStask task);
mal_export str MOSselect_runlength(Client cntxt,  MOStask task, void *low, void *hgh, bit *li, bit *hi, bit *anti);
mal_export str MOSthetaselect_runlength(Client cntxt,  MOStask task, void *val, str oper);
mal_export str MOSprojection_runlength(Client cntxt,  MOStask task);
mal_export str MOSjoin_runlength(Client cntxt,  MOStask task);
#endif /* _MOSAIC_RLE_ */
