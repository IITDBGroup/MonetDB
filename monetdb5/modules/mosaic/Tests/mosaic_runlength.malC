# OID list compressions

o := 19531025:int;

b:= bat.new(:int);
bat.append(b,50);
bat.append(b,50);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,o);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
bat.append(b,9);
b:= bat.append(b,b);
base:= algebra.copy(b);

io.print(base);
mosaic.compress(b,"runlength");
b:= mosaic.decompress(b);
io.print(b);
# OID list compressions


o := 19531025:int;

f:= bat.new(:flt);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.314);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.316);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);
bat.append(f,0.317);


io.print(f);
mosaic.compress(f,"runlength");
f:= mosaic.decompress(f);
io.print(f);

bs:= bat.new(:timestamp);
bat.append(bs,"2014-08-14 12:00:00":timestamp);
bat.append(bs,"2014-08-14 12:00:00":timestamp);
bat.append(bs,"2014-08-14 12:00:00":timestamp);
bat.append(bs,"2014-08-14 12:00:00":timestamp);
bat.append(bs,"2014-08-14 12:00:00":timestamp);
bat.append(bs,"2014-08-14 12:01:00":timestamp);
bat.append(bs,"2014-08-14 12:02:00":timestamp);
bs:= bat.append(bs,bs);

io.print(bs);
mosaic.compress(bs,"runlength");
bs:= mosaic.decompress(bs);
io.print(bs);

d:= bat.new(:date);
bat.append(d,"2014-08-14":date);
bat.append(d,"2014-08-14":date);
bat.append(d,"2014-08-14":date);
bat.append(d,"2014-08-14":date);
bat.append(d,"2014-08-15":date);
d:= bat.append(d,d);

io.print(d);
mosaic.compress(d,"runlength");
d:= mosaic.decompress(d);
io.print(d);


# handle compression of strings
s:= bat.new(:str);
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"hello world");
bat.append(s,"goodbye");
bat.append(s,"goodbye");
bat.append(s,"goodbye");
bat.append(s,"goodbye");
bat.append(s,"goodbye");

io.print(s);
mosaic.compress(s);
s:= mosaic.decompress(s);
io.print(s);
