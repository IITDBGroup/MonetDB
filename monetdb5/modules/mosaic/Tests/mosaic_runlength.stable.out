stdout of test 'mosaic_runlength` in directory 'monetdb5/modules/mosaic` itself:


# 20:55:33 >  
# 20:55:33 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37588" "--set" "mapi_usock=/var/tmp/mtest-15513/.s.monetdb.37588" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_rle.mal"
# 20:55:33 >  

# MonetDB 5 server v11.18.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.591 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:37588/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-15513/.s.monetdb.37588
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/R   module loaded

# 11:50:48 >  
# 11:50:48 >  "mclient" "-lmal" "-ftest" "-tnone" "-Eutf-8" "--host=/var/tmp/mtest-30332" "--port=39586"
# 11:50:48 >  

# 15:01:26 >  
# 15:01:26 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-15809" "--port=37660"
# 15:01:26 >  

#io.print(base);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	50  ]
[ 2@0,	19531025  ]
[ 3@0,	19531025  ]
[ 4@0,	19531025  ]
[ 5@0,	19531025  ]
[ 6@0,	19531025  ]
[ 7@0,	19531025  ]
[ 8@0,	19531025  ]
[ 9@0,	19531025  ]
[ 10@0,	19531025  ]
[ 11@0,	19531025  ]
[ 12@0,	19531025  ]
[ 13@0,	19531025  ]
[ 14@0,	19531025  ]
[ 15@0,	19531025  ]
[ 16@0,	19531025  ]
[ 17@0,	19531025  ]
[ 18@0,	19531025  ]
[ 19@0,	19531025  ]
[ 20@0,	19531025  ]
[ 21@0,	19531025  ]
[ 22@0,	9  ]
[ 23@0,	9  ]
[ 24@0,	9  ]
[ 25@0,	9  ]
[ 26@0,	9  ]
[ 27@0,	9  ]
[ 28@0,	9  ]
[ 29@0,	9  ]
[ 30@0,	9  ]
[ 31@0,	9  ]
[ 32@0,	9  ]
[ 33@0,	9  ]
[ 34@0,	9  ]
[ 35@0,	9  ]
[ 36@0,	9  ]
[ 37@0,	9  ]
[ 38@0,	9  ]
[ 39@0,	9  ]
[ 40@0,	9  ]
[ 41@0,	9  ]
[ 42@0,	9  ]
[ 43@0,	9  ]
[ 44@0,	9  ]
[ 45@0,	9  ]
[ 46@0,	9  ]
[ 47@0,	9  ]
[ 48@0,	9  ]
[ 49@0,	9  ]
[ 50@0,	9  ]
[ 51@0,	9  ]
[ 52@0,	50  ]
[ 53@0,	50  ]
[ 54@0,	19531025  ]
[ 55@0,	19531025  ]
[ 56@0,	19531025  ]
[ 57@0,	19531025  ]
[ 58@0,	19531025  ]
[ 59@0,	19531025  ]
[ 60@0,	19531025  ]
[ 61@0,	19531025  ]
[ 62@0,	19531025  ]
[ 63@0,	19531025  ]
[ 64@0,	19531025  ]
[ 65@0,	19531025  ]
[ 66@0,	19531025  ]
[ 67@0,	19531025  ]
[ 68@0,	19531025  ]
[ 69@0,	19531025  ]
[ 70@0,	19531025  ]
[ 71@0,	19531025  ]
[ 72@0,	19531025  ]
[ 73@0,	19531025  ]
[ 74@0,	9  ]
[ 75@0,	9  ]
[ 76@0,	9  ]
[ 77@0,	9  ]
[ 78@0,	9  ]
[ 79@0,	9  ]
[ 80@0,	9  ]
[ 81@0,	9  ]
[ 82@0,	9  ]
[ 83@0,	9  ]
[ 84@0,	9  ]
[ 85@0,	9  ]
[ 86@0,	9  ]
[ 87@0,	9  ]
[ 88@0,	9  ]
[ 89@0,	9  ]
[ 90@0,	9  ]
[ 91@0,	9  ]
[ 92@0,	9  ]
[ 93@0,	9  ]
[ 94@0,	9  ]
[ 95@0,	9  ]
[ 96@0,	9  ]
[ 97@0,	9  ]
[ 98@0,	9  ]
[ 99@0,	9  ]
[ 100@0,	9  ]
[ 101@0,	9  ]
[ 102@0,	9  ]
[ 103@0,	9  ]
#none 1
#none 2
#none 3
#none 4
#none 5
#none 6
#none 7
#none 8
#none 9
#none 10
#none 11
#none 12
#none 13
#none 14
#none 15
#none 16
#none 17
#none 18
#none 19
#none 20
#none 21
#none 22
#none 23
#none 24
#none 25
#none 26
#none 27
#none 28
#none 29
#none 30
#none 31
#none 32
#none 33
#none 34
#none 35
#none 36
#none 37
#none 38
#none 39
#none 40
#none 41
#none 42
#none 43
#none 44
#none 45
#none 46
#none 47
#none 48
#none 49
#none 50
#none 51
#none 52
#none 53
#none 54
#none 55
#none 56
#none 57
#none 58
#none 59
#none 60
#none 61
#none 62
#none 63
#none 64
#none 65
#none 66
#none 67
#none 68
#none 69
#none 70
#none 71
#none 72
#none 73
#none 74
#none 75
#none 76
#none 77
#none 78
#none 79
#none 80
#none 81
#none 82
#none 83
#none 84
#none 85
#none 86
#none 87
#none 88
#none 89
#none 90
#none 91
#none 92
#none 93
#none 94
#none 95
#none 96
#none 97
#none 98
#none 99
#none 100
#none 101
#none 102
#none 103
#none 104
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	50  ]
[ 2@0,	19531025  ]
[ 3@0,	19531025  ]
[ 4@0,	19531025  ]
[ 5@0,	19531025  ]
[ 6@0,	19531025  ]
[ 7@0,	19531025  ]
[ 8@0,	19531025  ]
[ 9@0,	19531025  ]
[ 10@0,	19531025  ]
[ 11@0,	19531025  ]
[ 12@0,	19531025  ]
[ 13@0,	19531025  ]
[ 14@0,	19531025  ]
[ 15@0,	19531025  ]
[ 16@0,	19531025  ]
[ 17@0,	19531025  ]
[ 18@0,	19531025  ]
[ 19@0,	19531025  ]
[ 20@0,	19531025  ]
[ 21@0,	19531025  ]
[ 22@0,	9  ]
[ 23@0,	9  ]
[ 24@0,	9  ]
[ 25@0,	9  ]
[ 26@0,	9  ]
[ 27@0,	9  ]
[ 28@0,	9  ]
[ 29@0,	9  ]
[ 30@0,	9  ]
[ 31@0,	9  ]
[ 32@0,	9  ]
[ 33@0,	9  ]
[ 34@0,	9  ]
[ 35@0,	9  ]
[ 36@0,	9  ]
[ 37@0,	9  ]
[ 38@0,	9  ]
[ 39@0,	9  ]
[ 40@0,	9  ]
[ 41@0,	9  ]
[ 42@0,	9  ]
[ 43@0,	9  ]
[ 44@0,	9  ]
[ 45@0,	9  ]
[ 46@0,	9  ]
[ 47@0,	9  ]
[ 48@0,	9  ]
[ 49@0,	9  ]
[ 50@0,	9  ]
[ 51@0,	9  ]
[ 52@0,	50  ]
[ 53@0,	50  ]
[ 54@0,	19531025  ]
[ 55@0,	19531025  ]
[ 56@0,	19531025  ]
[ 57@0,	19531025  ]
[ 58@0,	19531025  ]
[ 59@0,	19531025  ]
[ 60@0,	19531025  ]
[ 61@0,	19531025  ]
[ 62@0,	19531025  ]
[ 63@0,	19531025  ]
[ 64@0,	19531025  ]
[ 65@0,	19531025  ]
[ 66@0,	19531025  ]
[ 67@0,	19531025  ]
[ 68@0,	19531025  ]
[ 69@0,	19531025  ]
[ 70@0,	19531025  ]
[ 71@0,	19531025  ]
[ 72@0,	19531025  ]
[ 73@0,	19531025  ]
[ 74@0,	9  ]
[ 75@0,	9  ]
[ 76@0,	9  ]
[ 77@0,	9  ]
[ 78@0,	9  ]
[ 79@0,	9  ]
[ 80@0,	9  ]
[ 81@0,	9  ]
[ 82@0,	9  ]
[ 83@0,	9  ]
[ 84@0,	9  ]
[ 85@0,	9  ]
[ 86@0,	9  ]
[ 87@0,	9  ]
[ 88@0,	9  ]
[ 89@0,	9  ]
[ 90@0,	9  ]
[ 91@0,	9  ]
[ 92@0,	9  ]
[ 93@0,	9  ]
[ 94@0,	9  ]
[ 95@0,	9  ]
[ 96@0,	9  ]
[ 97@0,	9  ]
[ 98@0,	9  ]
[ 99@0,	9  ]
[ 100@0,	9  ]
[ 101@0,	9  ]
[ 102@0,	9  ]
[ 103@0,	9  ]
#--------------------------#
# h	t  # name
# void	flt  # type
#--------------------------#
[ 0@0,	0.314  ]
[ 1@0,	0.314  ]
[ 2@0,	0.314  ]
[ 3@0,	0.314  ]
[ 4@0,	0.314  ]
[ 5@0,	0.314  ]
[ 6@0,	0.314  ]
[ 7@0,	0.314  ]
[ 8@0,	0.314  ]
[ 9@0,	0.314  ]
[ 10@0,	0.314  ]
[ 11@0,	0.314  ]
[ 12@0,	0.314  ]
[ 13@0,	0.314  ]
[ 14@0,	0.314  ]
[ 15@0,	0.314  ]
[ 16@0,	0.314  ]
[ 17@0,	0.314  ]
[ 18@0,	0.314  ]
[ 19@0,	0.314  ]
[ 20@0,	0.314  ]
[ 21@0,	0.316  ]
[ 22@0,	0.316  ]
[ 23@0,	0.316  ]
[ 24@0,	0.316  ]
[ 25@0,	0.316  ]
[ 26@0,	0.316  ]
[ 27@0,	0.316  ]
[ 28@0,	0.316  ]
[ 29@0,	0.316  ]
[ 30@0,	0.316  ]
[ 31@0,	0.316  ]
[ 32@0,	0.316  ]
[ 33@0,	0.316  ]
[ 34@0,	0.316  ]
[ 35@0,	0.316  ]
[ 36@0,	0.316  ]
[ 37@0,	0.317  ]
[ 38@0,	0.317  ]
[ 39@0,	0.317  ]
[ 40@0,	0.317  ]
[ 41@0,	0.317  ]
[ 42@0,	0.317  ]
[ 43@0,	0.314  ]
[ 44@0,	0.314  ]
[ 45@0,	0.314  ]
[ 46@0,	0.314  ]
[ 47@0,	0.314  ]
[ 48@0,	0.314  ]
[ 49@0,	0.314  ]
[ 50@0,	0.314  ]
[ 51@0,	0.314  ]
[ 52@0,	0.314  ]
[ 53@0,	0.314  ]
[ 54@0,	0.314  ]
[ 55@0,	0.314  ]
[ 56@0,	0.314  ]
[ 57@0,	0.314  ]
[ 58@0,	0.314  ]
[ 59@0,	0.314  ]
[ 60@0,	0.314  ]
[ 61@0,	0.314  ]
[ 62@0,	0.314  ]
[ 63@0,	0.314  ]
[ 64@0,	0.314  ]
[ 65@0,	0.316  ]
[ 66@0,	0.316  ]
[ 67@0,	0.316  ]
[ 68@0,	0.316  ]
[ 69@0,	0.316  ]
[ 70@0,	0.316  ]
[ 71@0,	0.316  ]
[ 72@0,	0.316  ]
[ 73@0,	0.316  ]
[ 74@0,	0.316  ]
[ 75@0,	0.316  ]
[ 76@0,	0.316  ]
[ 77@0,	0.316  ]
[ 78@0,	0.316  ]
[ 79@0,	0.316  ]
[ 80@0,	0.316  ]
[ 81@0,	0.317  ]
[ 82@0,	0.317  ]
[ 83@0,	0.317  ]
[ 84@0,	0.317  ]
[ 85@0,	0.317  ]
[ 86@0,	0.317  ]
#--------------------------#
# h	t  # name
# void	flt  # type
#--------------------------#
[ 0@0,	0.314  ]
[ 1@0,	0.314  ]
[ 2@0,	0.314  ]
[ 3@0,	0.314  ]
[ 4@0,	0.314  ]
[ 5@0,	0.314  ]
[ 6@0,	0.314  ]
[ 7@0,	0.314  ]
[ 8@0,	0.314  ]
[ 9@0,	0.314  ]
[ 10@0,	0.314  ]
[ 11@0,	0.314  ]
[ 12@0,	0.314  ]
[ 13@0,	0.314  ]
[ 14@0,	0.314  ]
[ 15@0,	0.314  ]
[ 16@0,	0.314  ]
[ 17@0,	0.314  ]
[ 18@0,	0.314  ]
[ 19@0,	0.314  ]
[ 20@0,	0.314  ]
[ 21@0,	0.316  ]
[ 22@0,	0.316  ]
[ 23@0,	0.316  ]
[ 24@0,	0.316  ]
[ 25@0,	0.316  ]
[ 26@0,	0.316  ]
[ 27@0,	0.316  ]
[ 28@0,	0.316  ]
[ 29@0,	0.316  ]
[ 30@0,	0.316  ]
[ 31@0,	0.316  ]
[ 32@0,	0.316  ]
[ 33@0,	0.316  ]
[ 34@0,	0.316  ]
[ 35@0,	0.316  ]
[ 36@0,	0.316  ]
[ 37@0,	0.317  ]
[ 38@0,	0.317  ]
[ 39@0,	0.317  ]
[ 40@0,	0.317  ]
[ 41@0,	0.317  ]
[ 42@0,	0.317  ]
[ 43@0,	0.314  ]
[ 44@0,	0.314  ]
[ 45@0,	0.314  ]
[ 46@0,	0.314  ]
[ 47@0,	0.314  ]
[ 48@0,	0.314  ]
[ 49@0,	0.314  ]
[ 50@0,	0.314  ]
[ 51@0,	0.314  ]
[ 52@0,	0.314  ]
[ 53@0,	0.314  ]
[ 54@0,	0.314  ]
[ 55@0,	0.314  ]
[ 56@0,	0.314  ]
[ 57@0,	0.314  ]
[ 58@0,	0.314  ]
[ 59@0,	0.314  ]
[ 60@0,	0.314  ]
[ 61@0,	0.314  ]
[ 62@0,	0.314  ]
[ 63@0,	0.314  ]
[ 64@0,	0.314  ]
[ 65@0,	0.316  ]
[ 66@0,	0.316  ]
[ 67@0,	0.316  ]
[ 68@0,	0.316  ]
[ 69@0,	0.316  ]
[ 70@0,	0.316  ]
[ 71@0,	0.316  ]
[ 72@0,	0.316  ]
[ 73@0,	0.316  ]
[ 74@0,	0.316  ]
[ 75@0,	0.316  ]
[ 76@0,	0.316  ]
[ 77@0,	0.316  ]
[ 78@0,	0.316  ]
[ 79@0,	0.316  ]
[ 80@0,	0.316  ]
[ 81@0,	0.317  ]
[ 82@0,	0.317  ]
[ 83@0,	0.317  ]
[ 84@0,	0.317  ]
[ 85@0,	0.317  ]
[ 86@0,	0.317  ]
#--------------------------#
# h	t  # name
# void	timestamp  # type
#--------------------------#
[ 0@0,	2014-08-14 12:00:00.000000	]
[ 1@0,	2014-08-14 12:00:00.000000	]
[ 2@0,	2014-08-14 12:00:00.000000	]
[ 3@0,	2014-08-14 12:00:00.000000	]
[ 4@0,	2014-08-14 12:00:00.000000	]
[ 5@0,	2014-08-14 12:01:00.000000	]
[ 6@0,	2014-08-14 12:02:00.000000	]
[ 7@0,	2014-08-14 12:00:00.000000	]
[ 8@0,	2014-08-14 12:00:00.000000	]
[ 9@0,	2014-08-14 12:00:00.000000	]
[ 10@0,	2014-08-14 12:00:00.000000	]
[ 11@0,	2014-08-14 12:00:00.000000	]
[ 12@0,	2014-08-14 12:01:00.000000	]
[ 13@0,	2014-08-14 12:02:00.000000	]
#--------------------------#
# h	t  # name
# void	timestamp  # type
#--------------------------#
[ 0@0,	2014-08-14 12:00:00.000000	]
[ 1@0,	2014-08-14 12:00:00.000000	]
[ 2@0,	2014-08-14 12:00:00.000000	]
[ 3@0,	2014-08-14 12:00:00.000000	]
[ 4@0,	2014-08-14 12:00:00.000000	]
[ 5@0,	2014-08-14 12:01:00.000000	]
[ 6@0,	2014-08-14 12:02:00.000000	]
[ 7@0,	2014-08-14 12:00:00.000000	]
[ 8@0,	2014-08-14 12:00:00.000000	]
[ 9@0,	2014-08-14 12:00:00.000000	]
[ 10@0,	2014-08-14 12:00:00.000000	]
[ 11@0,	2014-08-14 12:00:00.000000	]
[ 12@0,	2014-08-14 12:01:00.000000	]
[ 13@0,	2014-08-14 12:02:00.000000	]
#--------------------------#
# h	t  # name
# void	date  # type
#--------------------------#
[ 0@0,	2014-08-14  ]
[ 1@0,	2014-08-14  ]
[ 2@0,	2014-08-14  ]
[ 3@0,	2014-08-14  ]
[ 4@0,	2014-08-15  ]
[ 5@0,	2014-08-14  ]
[ 6@0,	2014-08-14  ]
[ 7@0,	2014-08-14  ]
[ 8@0,	2014-08-14  ]
[ 9@0,	2014-08-15  ]
#--------------------------#
# h	t  # name
# void	date  # type
#--------------------------#
[ 0@0,	2014-08-14  ]
[ 1@0,	2014-08-14  ]
[ 2@0,	2014-08-14  ]
[ 3@0,	2014-08-14  ]
[ 4@0,	2014-08-15  ]
[ 5@0,	2014-08-14  ]
[ 6@0,	2014-08-14  ]
[ 7@0,	2014-08-14  ]
[ 8@0,	2014-08-14  ]
[ 9@0,	2014-08-15  ]
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"hello world"  ]
[ 1@0,	"hello world"  ]
[ 2@0,	"hello world"  ]
[ 3@0,	"hello world"  ]
[ 4@0,	"hello world"  ]
[ 5@0,	"hello world"  ]
[ 6@0,	"hello world"  ]
[ 7@0,	"hello world"  ]
[ 8@0,	"hello world"  ]
[ 9@0,	"hello world"  ]
[ 10@0,	"goodbye"  ]
[ 11@0,	"goodbye"  ]
[ 12@0,	"goodbye"  ]
[ 13@0,	"goodbye"  ]
[ 14@0,	"goodbye"  ]
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"hello world"  ]
[ 1@0,	"hello world"  ]
[ 2@0,	"hello world"  ]
[ 3@0,	"hello world"  ]
[ 4@0,	"hello world"  ]
[ 5@0,	"hello world"  ]
[ 6@0,	"hello world"  ]
[ 7@0,	"hello world"  ]
[ 8@0,	"hello world"  ]
[ 9@0,	"hello world"  ]
[ 10@0,	"goodbye"  ]
[ 11@0,	"goodbye"  ]
[ 12@0,	"goodbye"  ]
[ 13@0,	"goodbye"  ]
[ 14@0,	"goodbye"  ]

# 20:55:33 >  
# 20:55:33 >  "Done."
# 20:55:33 >  

